File: backend/ride-share/src/main/java/mk/ukim/finki/rideshare/web/response/RideResponse.java
package mk.ukim.finki.rideshare.web.response;

public record RideResponse(
        Long id,
        String origin,
        Double originLatitude,
        Double originLongitude,
        String destination,
        Double destinationLatitude,
        Double destinationLongitude,
        Boolean isDoorToDoor,
        String departureDate,
        String departureTime,
        Boolean isDepartureTimeFlexible,
        Double price,
        Boolean hasLuggageSpace,
        Integer capacity,
        Integer seatsLeft,
        Boolean isInstantBookingEnabled,
        String providerFullName
) {
}

File: frontend/ride-share/src/interfaces/response/RideResponse.ts
export interface RideResponse {
    id : number;
    origin: string;
    originLatitude: number;
    originLongitude: number;
    destination: string;
    destinationLatitude: number;
    destinationLongitude: number;
    isDoorToDoor: boolean;
    departureDate: string;
    departureTime: string;
    isDepartureTimeFlexible: boolean;
    price: number;
    hasLuggageSpace: boolean;
    capacity: number;
    seatsLeft: number;
    isInstantBookingEnabled: boolean;
    providerFullName: string;
}